server/server/src/metrics.rs

```diff
@@ -0,0 +1,215 @@
+use crate::keys::public_key;
+use crate::{keys, ServerError, ServerState};
+use lazy_static::lazy_static;
+use libsecp256k1::PublicKey;
+use lockbook_models::api::FileUsage;
+use lockbook_models::file_metadata::{EncryptedFileMetadata, FileType};
+use log::error;
+use prometheus::{register_int_gauge_vec, IntGaugeVec};
+use prometheus_static_metric::make_static_metric;
+use redis::{AsyncCommands, AsyncIter};
+use redis_utils::converters::JsonGet;
+use std::collections::HashSet;
+use std::fmt::Debug;
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+use uuid::Uuid;
+
+make_static_metric! {
+    pub struct MetricsStatistics: IntGauge {
+        "type" => {
+            total_users,
+            active_users,
+            total_documents,
+            total_document_bytes,
+        },
+    }
+}
+
+lazy_static! {
+    pub static ref METRICS_COUNTERS_VEC: IntGaugeVec = register_int_gauge_vec!(
+        "lockbook_metrics_counters",
+        "Lockbook's basic metrics of users and files derived from redis.",
+        &["type"]
+    )
+    .unwrap();
+    pub static ref METRICS_STATISTICS: MetricsStatistics =
+        MetricsStatistics::from(&METRICS_COUNTERS_VEC);
+}
+
+// # required metrics
+// X total users
+// active percent (just active users / total users in metrics.lockbook.com)
+// active users
+// new users
+// X total documents
+// X total bytes
+// premium users and percentage
+
+#[derive(Debug)]
+pub enum MetricsError {}
+
+// pub struct MetricsPreProcessingInfo<'a> {
+//     pub public_keys: Vec<PublicKey>,
+//     pub ids: HashMap<&'a PublicKey, Uuid>,
+//     pub metadatas: HashMap<&'a PublicKey, EncryptedFileMetadata>
+// }
+
+pub const TWO_DAYS_IN_MILLIS: u128 = 1000 * 60 * 60 * 24 * 2;
+
+pub fn start_metrics_worker(server_state: &Arc<ServerState>) {
+    let state_clone = server_state.clone();
+
+    tokio::spawn(async move {
+        if let Err(e) = start(state_clone).await {
+            error!("Metrics error: {:?}", e)
+        }
+    });
+}
+
+pub async fn start(server_state: Arc<ServerState>) -> Result<(), ServerError<MetricsError>> {
+    loop {
+        let public_keys = get_all_public_keys(&server_state).await?;
+        METRICS_STATISTICS.total_users.set(public_keys.len() as i64);
+
+        let mut total_documents = 0;
+        let mut total_bytes = 0;
+        let mut active_users = 0;
+
+        for public_key in public_keys {
+            let mut con = server_state.index_db_pool.get().await?;
+
+            let ids = get_users_owned_files(&mut con, &public_key).await?;
```

Comment:
```suggestion
            let ids = get_owned(&mut con, &public_key).await?;
```

everything else is implied by parameters and return values